Sources/PackageGraph/ResolvedProduct.swift

```diff
@@ -72,6 +89,40 @@ public final class ResolvedProduct {
         let recursiveDependencies = try targets.lazy.flatMap { try $0.recursiveTargetDependencies() }
         return Array(Set(targets).union(recursiveDependencies))
     }
+
+    private static func computePlatforms(targets: [ResolvedTarget]) -> SupportedPlatforms {
+        // merge from targets, preferring the max constraint
+        let declared = targets.reduce(into: [SupportedPlatform]()) { partial, item in
```

Comment:
Seems correct, only suggestion would be to use a private closure or function for the shared behavior.  Also, would it be easier to have a dictionary from platform to version since the order doesn't matter anyway (it gets sorted at the end).  In each iteration you'd just set the entry to the max(existing, new).  But those are just nits.

Reply:
done